### PR TITLE
Finish pspdfkit labs search

### PR DESCRIPTION
**Pull request by tmcw**
This branch addresses all of the current comments on #626: namely

* Fix rubocop issues
* Add changelog entry
* Remove unused tempfile entry
* Change --searchable to --unsearchable

The only thing that currently is tricky is that the `self.document` method is pushed from 20 to 21 lines, so it gets caught by rubocop. I see no clear way to make it shorter without making the code less direct, so happy to either adjust rubocop or take any approach the maintainers are happy with.

### CODE REVIEW

**Comment by tmcw** [comment]
Noting that tests are still failing: I didn't go through the dance of updating CI test fixtures. Looks like the steps to do that are in [CONTRIBUTING.md](https://github.com/realm/jazzy/blob/master/CONTRIBUTING.md#updating-the-integration-specs) and will require committing changes to the text fixture repo. (also, noting that to get the rake task to work i had to `gem install bacon` manually)

@captainbarbosa want to take the baton in this magical relay race of open source feature production?

**Comment by captainbarbosa** [comment]
@tmcw I've updated the integration specs per the instructions in [CONTRIBUTING.md](https://github.com/mapbox/jazzy/blob/master/CONTRIBUTING.md)

@jpsim I think this is ready to go, can we get push access so we can complete the final steps?

Also to note - while we were updating the integration spec, we had to update `MiscJazzyFeatures` Xcode project to explicitly set the `SWIFT_VERSION` to legacy:

```
=== BUILD TARGET MiscJazzyFeatures OF PROJECT MiscJazzyFeatures WITH THE DEFAULT CONFIGURATION (Release) ===

Check dependencies
“Use Legacy Swift Language Version” (SWIFT_VERSION) is required to be configured correctly for targets which use Swift. Use the [Edit > Convert > To Current Swift Syntax…] menu to choose a Swift version or use the Build Settings editor to configure the build setting directly.

** BUILD FAILED **
```

I also had to recursively clone the submodules, which wasn't mentioned in the instructions within [CONTRIBUTING.md](https://github.com/mapbox/jazzy/blob/master/CONTRIBUTING.md) - I can post a separate PR to include this.

**Comment by jpsim** [comment]
Oh hai mapbox! 👋 

Thanks for pushing this forward! And of course thanks to @esad for the initial implementation.

> I also had to recursively clone the submodules, which wasn't mentioned in the instructions within CONTRIBUTING.md - I can post a separate PR to include this.

Oh, that'd be great! Yes please.

I've just granted @captainbarbosa and @tmcw push access to both this repo and the integration specs repo.

File: lib/jazzy/config.rb [hunk 1]
```diff
@@ -246,6 +246,11 @@ def expand_path(path)
                    'https://github.com/realm/realm-cocoa/tree/v0.87.1)'
 
     # ──────── Doc generation options ────────
+    config_attr :unsearchable,
```
Comment (jpsim) [inline lib/jazzy/config.rb]: I think this should be named either `disable_search`/`--disable-search` or `skip_search`/`--skip-search` to match the other options.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,7 +6,8 @@
 
 ##### Enhancements
 
-* None.
+* By default, docs are now searchable. A new option, `--unsearchable`, lets you
```
Comment (jpsim) [inline CHANGELOG.md]: Only the `fullwidth` theme appears to be searchable. Also, this should be reformatted to match the standard changelog entry format defined in https://github.com/realm/jazzy/blob/master/CONTRIBUTING.md#tracking-changes

With credit given to @esad, @tmcw and @captainbarbosa

File: lib/jazzy/themes/fullwidth/assets/js/lunr.min.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/**
+ * lunr - http://lunrjs.com - A bit like Solr, but much smaller and not as bright - 0.7.0
```
Comment (jpsim) [inline lib/jazzy/themes/fullwidth/assets/js/lunr.min.js]: Should we update this to 0.7.2? https://github.com/olivernn/lunr.js/releases/tag/v0.7.2

**Comment by jpsim** [comment]
Would you like me to address the remaining items, or should I leave that to you @tmcw?

**Comment by captainbarbosa** [comment]
@jpsim I made the changes as advised from your review - let me know if there's anything else I can do!

**Comment by jpsim** [comment]
Thanks @captainbarbosa. There's now a merge conflict with the changelog and the integration specs need to be rebuilt again.

**Comment by friedbunny** [comment]
Here’s the rubocop failure @tmcw mentioned:

```
lib/jazzy/doc_builder.rb:322:5: C: Method has too many lines. [21/20]
    def self.document(source_module, doc_model, path_to_root) ...
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

25 files inspected, 1 offense detected
RuboCop failed!
```

**Comment by friedbunny** [comment]
@jpsim We’re having some issues regenerating the integration specs. The new files from this PR are successfully added, but the docs generated for Siesta are inexplicably having sections removed:

```diff
diff --git a/document_siesta/after/api-docs/Classes/Resource.html b/document_siesta/after/api-docs/Classes/Resource.html
index 1c574ce..0fd0aee 100644
--- a/document_siesta/after/api-docs/Classes/Resource.html
+++ b/document_siesta/after/api-docs/Classes/Resource.html
@@ -740,29 +740,6 @@ up to you to canonicalize your parameter order.</p>
 <p>Note that, unlike load() and loadIfNeeded(), this method does <em>not</em> update latestData or latestError,
 and does not notify resource observers about the result.</p>
 
-<div class="aside aside-parameter">
-    <p class="aside-title">Parameter</p>
-    Parameter method: The HTTP verb to use for the request
-
-</div>
-
-<div class="aside aside-parameter">
-    <p class="aside-title">Parameter</p>
-    <p>Parameter requestMutation:
-An optional callback to change details of the request before it is sent. For example:</p>
-<pre class="highlight plaintext"><code>request(.POST) { nsreq in
-  nsreq.HTTPBody = imageData
-  nsreq.addValue(
-    "image/png",
-    forHTTPHeaderField:
-      "Content-Type")
-}
-</code></pre>
-
-<p>Does nothing by default.</p>
-
-</div>
-
```

_[Full diff of the spec regeneration is available here.](https://gist.github.com/friedbunny/f55dc3214cec0b9f8a785bfe96c69fdd)_

Siesta does not use the full-width theme and I therefore wouldn’t expect any changes to be made. `document_siesta/after/execution_output.txt` [has only one lined changed](https://gist.github.com/friedbunny/f55dc3214cec0b9f8a785bfe96c69fdd#file-jazzy-spec-regeneration-issue-diff-L481-L490) and that one is expected: “building search index”.

[Here’s the output from `bundle exec rake rebuild_integration_fixtures`.](https://gist.github.com/friedbunny/f9199316bab1d2822f32a61ee5490706) The relevant Siesta lines don’t have much information:

```
    Creates Siesta docs
      ✓ $ jazzy --output api-docs
      - execution_output.txt [FAILED]
      ✓ should not produce unexpected files
```

**Comment by friedbunny** [comment]
The above omissions do not happen when using 0.7.3, but still happen on this branch even when the `--disable-search` flag is used. 😞

**Comment by jpsim** [comment]
Yeah, the missing Siesta declaration is definitely a regression in this branch. I'm looking into it now.

**Comment by jpsim** [comment]
I'm guessing at this point that `SearchBuilder.build(...)` somehow mutates `sourcemodule` in such a way to remove the asides used by Siesta, but I don't see _how_ it could actually do that.

**Comment by 1ec5** [comment]
The Siesta deletions aren't new: they started happening shortly after the latest release, but before my recent PRs: https://github.com/realm/jazzy/pull/702#issuecomment-268688132

Do the integration specs pull in the latest version of the Siesta package, which could've changed, or does this necessarily point to a regression in jazzy?

**Comment by jpsim** [comment]
I think these specific siesta deletions are different, as running with the following patch preserves those asides:

```diff
--- a/lib/jazzy/doc_builder.rb
+++ b/lib/jazzy/doc_builder.rb
@@ -125,11 +125,6 @@ module Jazzy
       output_dir = options.output
       build_docs(output_dir, source_module.docs, source_module)
 
-      unless options.disable_search
-        warn 'building search index'
-        SearchBuilder.build(source_module, output_dir)
-      end
-
       copy_assets(output_dir)
 
       DocsetBuilder.new(output_dir, source_module).build!
```

**Comment by jpsim** [comment]
Actually, that's not even it. If I just run the siesta specs with this branch, the asides are also preserved. They just disappear if I run the whole test suite. Leading me now to think that it's some other mutation that's causing this...

**Comment by friedbunny** [comment]
The vanishing Siesta asides bisected to https://github.com/realm/jazzy/pull/697 — the integration spec apparently wasn’t updated properly for that PR. I’ve proposed https://github.com/realm/jazzy-integration-specs/pull/25 to fix the issue, which simply accepts the deletions as proper.

**Comment by friedbunny** [comment]
In 7825086, I went ahead and disabled the rubocop method length failure from  https://github.com/realm/jazzy/pull/703#issuecomment-271035374.

Please let me know if there’s anything I’m missing, but I believe this is what’s left to do:

- [ ] Merge https://github.com/realm/jazzy-integration-specs/pull/25.
- [ ] Push rebuilt integration specs for this PR to realm/jazzy-integration-specs branch.
- [ ] Update integration specs submodule commit hash in this PR.
- [ ] Merge rebuilt integration specs.
- [ ] 🥃

Excited to get this train into the station! 🙇

**Comment by SDGGiesbrecht** [comment]
Thanks for fixing the specs.

**Comment by jpsim** [comment]
Your prognosis and fixes look right to me @friedbunny! Thanks for everything. Let's get this green by pushing the rebuilt integration specs and pointing this branch to it.

**Comment by friedbunny** [comment]
Awesome! I don’t have push access, but I was going to tag @captainbarbosa in again, anyway. 😁

**Comment by jpsim** [comment]
I gave you write access in case that helps moving forward.

**Comment by jpsim** [comment]
You got so close! Will finish finishing this finishment in #725.

**Pull request closed without merging**